tests/unit/auth/oidc.ts

```diff
@@ -0,0 +1,91 @@
+import { createClient, rqlBuilder } from '../../../src';
+import { createProviderData, generateRandomName } from '../../__helpers__/oidc';
+import { OidcProviderResponse } from '../../../src/models/oidc/Providers';
+
+describe('Enables an OpenID Connect workflow', () => {
+  let provider: OidcProviderResponse;
+
+  const sdk = createClient({
+    host: process.env.API_HOST,
+    consumerKey: process.env.CONSUMER_KEY,
+    consumerSecret: process.env.CONSUMER_SECRET,
+  });
+
+  beforeAll(async () => {
+    await sdk.auth.authenticate({
+      email: process.env.API_USERNAME,
+      password: process.env.API_PASSWORD,
+    });
+  });
+
+  beforeEach(async () => {
+    const data = createProviderData();
```

Comment:
This technically would be nicer if it was declared individually in each test, especially the create. But it seemed unnecessary.